iree/compiler/Conversion/LinalgToSPIRV/LinalgTileAndFusePass.cpp

```diff
@@ -43,16 +44,15 @@ namespace iree_compiler {
 // Utility functions
 //===----------------------------------------------------------------------===//
 
-static ArrayRef<int64_t> dropTrailingOnes(ArrayRef<int64_t> vector) {
-  if (vector.empty()) return vector;
-  auto numTrailingOnes = 0;
-  for (unsigned i = vector.size() - 1; i > 0; --i) {
-    if (vector[i] != 1) {
-      break;
-    }
-    numTrailingOnes++;
+/// Returns the workgroup size currently set on the function that the current
+/// `operation` is in.
+static std::array<int64_t, 3> getWorkgroupSizeFromAttr(Operation *op) {
+  std::array<int64_t, 3> workgroupSize = {1, 1, 1};
+  if (DenseIntElementsAttr attr = spirv::lookupLocalWorkGroupSize(op)) {
+    for (auto val : enumerate(attr.getValues<APInt>()))
+      workgroupSize[val.index()] = val.value().getSExtValue();
```

Comment:
Do we need to check/assert the size of attr.getValues is as same as `3` (use workgropuSize.size() would be better).